### PR TITLE
Claiming a talk

### LINKED ISSUE #80: User: Claiming a talk
```
@talk
Feature:
  As a user
  In order to connect me to a talk
  I need to claim a talk

  Background:
    Given there is a "Local meetup" organization
    And "March 2019 gathering" is scheduled for "2019-03-07"
    And there is a talk "Something about nothing"

  Scenario: Speaker can claim unclaimed talk
    Given I'am logged in as "Jo Johnson"
    When I claim "Something about nothing"
    Then I should have "Something about nothing" claimed

  Scenario: Talks with pending claims can not be claimed
    Given "Something about nothing" has a pending claim by "Jo Johnson"
    And I'am logged in as "Alex Smith"
    When I claim "Something about nothing"
    Then I should see an error saying there is a pending claim on the talk

  Scenario: Talks with rejected claims can be claimed
    Given "Something about nothing" has a rejected claim by "Jo Johnson"
    And I'am logged in as "Alex Smith"
    When I claim "Something about nothing"
    Then I should have "Something about nothing" claimed

  Scenario: Talks with speaker assigned can not be claimed
    Given "Something about nothing" has "Jo Johnson" assigned as speaker
    And I'am logged in as "Alex Smith"
    When I claim "Something about nothing"
    Then I should see an error saying there speaker is already assigned
```

### PR DESCRIPTION
**Pull request by msvrtan**
This PR covers claiming a talk by user.

I would suggest you all to go commit-by-commit as contexts and code change as I implemented each scenario, changing the code structure of previous ones

Closes #80

### CODE REVIEW

File: src/Talk/Entity/TalkEntity.php [hunk 1]
```diff
@@ -37,6 +42,30 @@ public function __construct(
         $this->title       = $title;
         $this->description = $description;
         $this->speakerName = $speakerName;
+        $this->claims      = [];
+    }
+
+    public function claimTalk(UserEntity $speaker)
+    {
+        if (null !== $this->speaker) {
```
Comment (leovujanic) [inline src/Talk/Entity/TalkEntity.php]: I think you could move this check to `hasSpeaker()` method. It's defined already but not implemented.

File: src/Talk/Entity/TalkEntity.php [hunk 2]
```diff
@@ -37,6 +42,30 @@ public function __construct(
...
+            throw new \DomainException('Talk already has a speaker set');
+        }
+
+        foreach ($this->claims as $claim) {
```
Comment (leovujanic) [inline src/Talk/Entity/TalkEntity.php]: What do you think? Can we maybe move this logic to separate method `hasPendingClaim()` to reduce the responsibility of this method and for future use?

File: src/Talk/Entity/TalkEntity.php (hunk 1 above)
Comment (msvrtan) [reply src/Talk/Entity/TalkEntity.php]: Damn, missed that one ! 

Fixed

File: src/Talk/Entity/TalkEntity.php (hunk 2 above)
Comment (msvrtan) [reply src/Talk/Entity/TalkEntity.php]: 👍 Fixed

File: src/Organization/Entity/ClaimEntity.php [hunk 3]
```diff
@@ -40,4 +40,13 @@ public function markAsRejected()
     {
         $this->approved = false;
     }
+
+    public function isPending()
+    {
+        if (null === $this->approved) {
```
[the remark this answers is not in the document]
Comment (msvrtan) [reply src/Organization/Entity/ClaimEntity.php]: IMO, it has better readability ... cause I dont have to spend extra grey cells to decypher when it's gonna be false or true (but again, that's me :) )

[the remark this answers is not in the document]
Comment (leovujanic) [reply src/Organization/Entity/ClaimEntity.php]: Perfectly valid

**Pull request merged**